src/librustc_mir/const_eval/machine.rs

```diff
@@ -345,11 +346,19 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter {
         Ok(())
     }
 
-    fn before_access_static(
+    fn before_access_global(
         memory_extra: &MemoryExtra,
-        _allocation: &Allocation,
+        alloc_id: AllocId,
+        allocation: &Allocation,
+        def_id: Option<DefId>,
```

Comment:
Maybe rename this to `static_def_id`? (in a future PR, since this one is in a rollup)